From 56b0c8ac846c18002e32e93629ae04a0a6281706 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Mon, 26 Sep 2022 20:52:54 +0100
Subject: [PATCH 1/3] Fix libcxx build without <tr1/*> (use <memory> if
 available)

With Clang & libcxx 15, tr1/* doesn't seem to be available,
which is fine, as we can use the proper <memory> header
instead.

We already do this in several other locations. Fall back
to tr1 and Boost as before.

Signed-off-by: Sam James <sam@gentoo.org>
Ref: https://bugs.gentoo.org/739034
Ref: https://bugs.gentoo.org/715514
Origin: Gentoo
Upstream-Status: Merged [https://codeberg.org/shelter/reschroot/pulls/5]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 bin/schroot-base/schroot-base-options.h |    4 +++-
 bin/schroot/schroot-options-base.h      |    4 +++-
 sbuild/sbuild-basic-keyfile.tcc         |    2 +-
 sbuild/sbuild-feature.cc                |    1 +
 4 files changed, 8 insertions(+), 3 deletions(-)

--- a/bin/schroot-base/schroot-base-options.h
+++ b/bin/schroot-base/schroot-base-options.h
@@ -27,7 +27,9 @@
 #include <string>
 #include <stdexcept>
 
-#ifdef HAVE_TR1_MEMORY
+#ifdef HAVE_MEMORY_SHARED_PTR
+#include <memory>
+#elif HAVE_TR1_MEMORY
 #include <tr1/memory>
 #elif HAVE_BOOST_SHARED_PTR_HPP
 #include <boost/shared_ptr.hpp>
--- a/bin/schroot/schroot-options-base.h
+++ b/bin/schroot/schroot-options-base.h
@@ -27,7 +27,9 @@
 #include <string>
 #include <vector>
 
-#ifdef HAVE_TR1_MEMORY
+#ifdef HAVE_MEMORY_SHARED_PTR
+#include <memory>
+#elif HAVE_TR1_MEMORY
 #include <tr1/memory>
 #elif HAVE_BOOST_SHARED_PTR_HPP
 #include <boost/shared_ptr.hpp>
-- 
Fixed up by sm-checkpatch 0.1

